src/check_plugin_vulnerability.py

```diff
@@ -28,12 +28,20 @@
 import requests
 from sys import exit
 
-SCRIPT = (
-    'def warnings_monitor = new jenkins.security.UpdateSiteWarningsMonitor()\n'  # NOQA E501 Don't wrap because of groovy code
-    'def plugin_vulnerabilities = warnings_monitor.getActivePluginWarningsByPlugin()\n'  # NOQA E501 Don't wrap because of groovy code
-    'def critical_plugins = plugin_vulnerabilities.keySet()*.longName\n'
-    'print(new groovy.json.JsonBuilder(critical_plugins))'
-)
+SCRIPT = """
+    import jenkins.security.*
+    ExtensionList<UpdateSiteWarningsConfiguration> configurations = ExtensionList.lookup(UpdateSiteWarningsConfiguration.class);
+    UpdateSiteWarningsConfiguration configuration = configurations.get(0);
+    problematic_plugins = []
+    current_warnings = configuration.getApplicableWarnings()
+    current_warnings.each {
+        if (it.type.name() == 'PLUGIN') {
+            problematic_plugins.add(it.component)
+        }
+    }
```

Comment:
You can also write it like that:
```suggestion
    def problematic_plugins = current_warnings.findAll { it.type.name() == 'PLUGIN') }.unique()
```

Reply:
Which is less readable imho. I'll pass.